### PR TITLE
Change build to update CUTEst on linux

### LINKED ISSUE #129: Make `Pkg.build` reinstall cutest/sifdecode/mastsif
Better would be check if a reinstall is needed, but this would be OS specific.

Related to #128.

### PR DESCRIPTION
**Pull request by abelsiqueira**
Closes #129

### CODE REVIEW

**Comment by abelsiqueira** [comment]
@dpo I'm not sure if the updating is working for OSX. Otherwise, should be good to go.

File: deps/build.jl [hunk 1]
```diff
@@ -54,21 +54,26 @@ end
 here = dirname(@__FILE__)
 cutestenv = joinpath(here, "cutestenv.jl")
 
+# Check if there is an external CUTEst installation
 if validate_libcutest()
+  cutest_path = get(ENV, "CUTEST", "")
+  info("External CUTEst installation found on $cutest_path")
```
Comment (dpo) [inline deps/build.jl]: Shouldn't there be a check that `cutest_path` is nonempty?

File: deps/build.jl [hunk 2]
```diff
@@ -91,29 +96,32 @@ else
...
     end
+  elseif is_linux()
+    cd(here) do
+      isdir("files") || mkdir("files")
+      cd("files") do
+        if isfile("install.sh")
+          info("Updating CUTEst")
+        else
+          info("Installing CUTEst")
+        end
```
Comment (dpo) [inline deps/build.jl]: Is this `if` necessary? The action is the same in both cases.

File: deps/build.jl (hunk 1 above)
Comment (abelsiqueira) [inline deps/build.jl]: I was thinking that `validate_libcutest` would only work if `CUTEST` existed from another installation, but I see that is not exactly the case.

**Comment by abelsiqueira** [comment]
After #132, the problems that were randomly happening shouldn't happen anymore.

**Comment by dpo** [comment]
What happens anyways when you say `Pkg.update()`? Is a `Pkg.build()` triggered?

File: deps/build.jl [hunk 3]
```diff
@@ -54,21 +54,25 @@ end
...
-    if validate_libcutest()
-      println("Valid CUTEst installation detected, no files installed.\nTo force installation, delete $cutestenv and run Pkg.build(\"CUTEst\") again.")
-      install = false
+  @static if is_apple()
+    install = true
+    if isfile(cutestenv)
+      include(cutestenv)
+      if validate_libcutest()
+        info("Updating CUTEst")
+        Homebrew.brew(`update cutest`)
```
Comment (dpo) [inline deps/build.jl]: Should probably also update mastsif here.

File: deps/build.jl [hunk 4]
```diff
@@ -91,29 +95,29 @@ else
...
-    @static if is_linux()
-      cd(here) do
-        isdir("files") && rm("files", recursive=true)
-        mkdir("files")
-        cd("files") do
-          lnxurl = "https://raw.githubusercontent.com/abelsiqueira/linux-cutest/v0.2.1/install.sh"
-          run(`wget $lnxurl -O install.sh`)
-          ENV["C_INCLUDE_PATH"] = joinpath(here, "usr", "include")
-          run(`bash install.sh`)
+        lnxurl = "https://raw.githubusercontent.com/abelsiqueira/linux-cutest/v0.3/install.sh"
```
Comment (dpo) [inline deps/build.jl]: I tagged v0.4 last night.

**Comment by abelsiqueira** [comment]
From what I saw [here](https://docs.julialang.org/en/stable/stdlib/pkg/#Base.Pkg.build), `update` calls `resolve`, that calls `build`. But I haven't tested.

**Comment by abelsiqueira** [comment]
Done

**Comment by dpo** [comment]
Thank you!

**Pull request merged**